example/example/spiders/books_06.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""
+Scrapy spider with reusable components used for two types of pages:
+* Listings page, with the components: book list and pagination
+* Detail book page, with the components: the item itself, the breadcrumbs and
+  the recently viewed list of books
+
+All links to detail book pages are found are followed.
+All links to search result pages are followed as well.
+"""
+
+import scrapy
+import attr
+
+from scrapy_po import WebPage, ItemPage
+
+
+class ListingsPiece(WebPage):
+    def urls(self):
+        return self.css('.image_container a::attr(href)').getall()
+
+
+class PaginationPiece(WebPage):
+    def urls(self):
+        return self.css('.pager a::attr(href)').getall()
+
+
+class BreadcrumbsPiece(WebPage):
+    def urls(self):
+        return self.css('.breadcrumb a::attr(href)').getall()
+
+
+@attr.s(auto_attribs=True)
+class ListingsPage(WebPage):
+    book_list: ListingsPiece
+    pagination: PaginationPiece
+
+
+@attr.s(auto_attribs=True)
+class BookPage(WebPage, ItemPage):
+    recently_viewed: ListingsPiece
```

Comment:
This is the part I'm not sure about, as a general pattern. There can be several listings on a single page - e.g. 

* recently viewed books 
* suggested/recommended books
* reviews for this book

It works on this particular website, but as an example it may not scale. Because of that I'd remove ListingPiece from the example, and add a method to BookPage instead for now.

Reply:
Ok, I changed that way by now: https://github.com/scrapinghub/scrapy-po/pull/5/commits/4ed02bbc5b13f2e032a982f6b8b10e8171e26120

But once we change the spiders for Scrapy 2.0 we can rewrite it with the `follow_all`. Probably better to address this change of examples to Scrapy 2.0 in a different PR to simplify the approval of this one.